src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPage.tsx

```diff
@@ -46,14 +46,17 @@ const ActionDetailsPage = () => {
   const isValidTx = isTransactionFormat(transactionHash);
   const events = ['event']; // to be taken from real data
 
-  const { data: colonyData, loading: loadingColony } =
-    useGetFullColonyByAddressQuery({
-      variables: {
-        address: action?.colonyAddress ?? '',
-      },
-      skip: !action?.colonyAddress,
-    });
-  const txColony = colonyData?.getColonyByAddress?.items[0];
```

Comment:
I think this query was redundant, we can check whether the action belongs to the current colony using the object from colony context. It made the details page load much faster